src/utils/word-api.ts

```diff
@@ -160,6 +214,65 @@ class WordApi {
       }
     });
   }
+
+  static async removeCurrentCitation(): Promise<unknown> {
+    return Word.run(async (context) => {
+      context.document
+        .getSelection()
+        .contentControls.getFirstOrNullObject()
+        .delete(false);
+      return context.sync();
+    }).catch((error) => {
+      console.log(`Error: ${JSON.stringify(error)}`);
+      if (error instanceof OfficeExtension.Error) {
+        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
+      }
+    });
+  }
+
+  static addEventListener(eventHandler: () => Promise<void>): void {
+    return Office.context.document.addHandlerAsync(
+      Office.EventType.DocumentSelectionChanged,
+      eventHandler,
+      (result) => {
+        console.log(`result: ${JSON.stringify(result)}`);
+      }
+    );
+  }
+
+  static removeEventListener(): void {
+    return Office.context.document.removeHandlerAsync(
+      Office.EventType.DocumentSelectionChanged,
+      (result) => {
+        console.log(`result: ${JSON.stringify(result)}`);
+      }
+    );
+  }
+
+  static async getItemsInCurrentSelection(): Promise<Array<string> | void> {
```

Comment:
I'm a bit undecided about this, but I would propose to remove the `void` case (e.g. by returning an empty array here) in case of an error. You ignore the void return value anyway by explicit casting, so there is no value in having it. Would also reduce the need for further changes once we have a nice global error handling. This remark applies to essentially all methods that were added.

Reply:
Something like this ?


`async getItemsInSelectedCitation(): Promise<Array<string>> {`
`    return Word.run(async (context: Word.RequestContext) => {`
`       Some code ...`
`    }).catch((error) => {`
`        Some code...`
`      return [];`
`    });`
`  }`

Reply:
Yes that looks better

Reply:
Yes, that's what I meant. Please also change the other methods accordingly, and remove the explicit `as` casts which are then no longer necessary.